org.uqbar.project.wollok.game/src/org/uqbar/project/wollok/game/gameboard/Gameboard.xtend

```diff
@@ -0,0 +1,117 @@
+package org.uqbar.project.wollok.game.gameboard;
+
+import com.badlogic.gdx.backends.lwjgl.LwjglApplication
+import java.util.ArrayList
+import java.util.Collection
+import java.util.List
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.uqbar.project.wollok.game.GameboardFactory
+import org.uqbar.project.wollok.game.Image
+import org.uqbar.project.wollok.game.Position
+import org.uqbar.project.wollok.game.VisualComponent
+import org.uqbar.project.wollok.game.listeners.ArrowListener
+import org.uqbar.project.wollok.game.listeners.GameboardListener
+import org.uqbar.project.wollok.interpreter.core.WollokProgramExceptionWrapper
+
+@Accessors
+class Gameboard {
+
+	public static Gameboard instance;
+	public static final int CELLZISE = 50
+	private VisualComponent character
+	private List<Cell> cells = new ArrayList<Cell>()
+	
+	public String title
+	public int height
+	public int width
+	public List<VisualComponent> components = new ArrayList<VisualComponent>()
+	public List<GameboardListener> listeners = new ArrayList<GameboardListener>()
+	
+	def static getInstance() {
+		if (instance == null) {
+			instance = new Gameboard()
+			GameboardFactory.setGame(instance)
+		}
+		return instance
+	}
+
+	def void start() {
+		new LwjglApplication(new GameboardRendering(this), new GameboardConfiguration(this));
+	}
+	
+	def void draw(Window window) {
+		// NO UTILIZAR FOREACH PORQUE HAY UN PROBLEMA DE CONCURRENCIA AL MOMENTO DE VACIAR LA LISTA
+		for (var i=0; i < this.listeners.size(); i++){
```

Comment:
This is a classic issue with listeners. Another option to still use "foreach" is to clone the list first.
Actually one should always do that in observers. You need to notify a cloned version of the list, in case while notifying one listener wants to stop listening.